apps/browser/qml/pages/components/ConfigDialog.qml

```diff
@@ -98,26 +99,50 @@ Dialog {
             id: loader
             height: Theme.itemSizeMedium
             width: prefsList.width
-            sourceComponent: model.type === 128 ? textSwitch : textField
+            sourceComponent: model.type == WebEngineSettings.BoolPref ? textSwitch : textField
 
             Component {
                 id: textField
 
                 TextField {
-                    label: model.name
+                    readonly property bool digitsOnly: model.type == WebEngineSettings.IntPref
+
+                    label: {
+                        if (errorHighlight && digitsOnly) {
+                            if (text.charAt(0) == "-") {
+                                //% "Select a value that is no less than -2147483648"
+                                return qsTrId("sailfish_browser-la-min_int_warning")
+                            }
+
+                            //% "Select a value that is no more than 2147483647"
+                            return qsTrId("sailfish_browser-la-max_int_warning")
+                        }
+                        return model.name
+                    }
                     text: model.value
                     placeholderText: model.name
-                    inputMethodHints: (model.type == 64 ? Qt.ImhDigitsOnly : 0)
+                    inputMethodHints: (digitsOnly ? Qt.ImhDigitsOnly : 0)
                     width: parent.width
                     height: Theme.itemSizeMedium
 
+                    Component.onCompleted: {
+                        if (digitsOnly)
+                            validator = intValidator
+                    }
+
                     onTextChanged: {
                         if (text === model.value) return;
 
-                        if (model.type == 64) {
-                            changedConfigs[model.name] = parseInt(text);
+                        if (digitsOnly) {
+                            changedConfigs[model.name] = {
+                                value: parseInt(text),
```

Comment:
I wonder if it's worth forcing to base 10, to avoid zero prefixes getting interpreted as octals.

Reply:
Shouldn't harm